test/unit/puppetlabs/puppetserver/lockable_pool_test.clj

```diff
@@ -235,6 +235,11 @@
       @blocked-borrow-thread-started?
       @lock-thread-started?
       (is (not (realized? blocked-borrow-thread-borrowed?)))
+      (let [start (System/currentTimeMillis)]
+        (while (and (not (.isLocked pool))
+                    (< (- (System/currentTimeMillis) start) 10000))
+          (Thread/yield)))
+      (is (.isLocked pool))
       (is (not (realized? lock-acquired?)))
 
       (return-instances pool instances)
```

Comment:
>We had a discussion at one time about changing the name of .isLocked to something else because of concerns about whether this is really fulfilling the definition of "locked" in that it returns true between the phases of "caller has been established as the next lock holder" and "currently borrowed instances have been checked back into the pool". Still open to changing that name at some point but probably outside of the scope of this PR.

In trying to understand these test changes (and having not actually fully read/parsed this comment), I was really confused by this naming. I think I would be +1 on a name change, but agree that it's outside the scope of this PR.